lms/djangoapps/commerce/views.py

```diff
@@ -0,0 +1,179 @@
+""" Commerce views. """
+
+import logging
+from simplejson import JSONDecodeError
+
+from django.conf import settings
+from django.utils.translation import ugettext as _
+import jwt
+from opaque_keys.edx.keys import CourseKey
+import requests
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.status import HTTP_406_NOT_ACCEPTABLE, HTTP_503_SERVICE_UNAVAILABLE, HTTP_202_ACCEPTED, HTTP_200_OK
+
+from rest_framework.views import APIView
+
+from commerce.constants import OrderStatus
+from course_modes.models import CourseMode
+from courseware import courses
+from student.models import CourseEnrollment
+from util.json_request import JsonResponse
+
+
+log = logging.getLogger(__name__)
+
+
+class DetailResponse(JsonResponse):
+    """ JSON response that simply contains a detail field. """
+
+    def __init__(self, message, status=HTTP_200_OK):
+        data = {'detail': message}
+        super(DetailResponse, self).__init__(object=data, status=status)
+
+
+class ApiErrorResponse(DetailResponse):
+    """ Response returned when calls to the E-Commerce API fail or the returned data is invalid. """
+
+    def __init__(self):
+        message = _('Call to E-Commerce API failed. Order creation failed.')
```

Comment:
Our REST API conventions suggest returning both a developer and a user-facing message (translated) in error responses. Do you think that's appropriate here?